### DISCUSSION TITLE
How to make `ovm-allowances.json` of op-migrate

### DISCUSSION THREAD

**Discussion by theo-learner** [ethereum-optimism/developers#22] [Miscellaneous] [answered]
### Issue Description

Hi all. I'm trying to reproduce the Bedrock migration in the article https://blog.oplabs.co/reproduce-bedrock-migration/. The op-migrate uses 3 inputs: ovm-addresses.json, ovm-messages.json, ovm-allowances.json. but I don't know how to make ovm-allowances.json. None of the source code in the optimism-legacy repository was for generating ovm-allowance.json. Anyone who know it?

repository: https://github.com/ethereum-optimism/optimism-legacy
migration command of op-migrate:
```
go run cmd/op-migrate/main.go \
 --l1-rpc-url "$L1_RPC" \
 --ovm-addresses $INPUT_DATA/ovm-addresses.json \
 --ovm-allowances $INPUT_DATA/ovm-allowances.json \
 --ovm-messages $INPUT_DATA/ovm-messages.json \
 --witness-file $DB_PATH/l2geth-state \
 --db-path $DB_PATH \\
 --deploy-config $DEPLOY_CONFIG \
 --network mainnet \
 --hardhat-deployments $DEPLOYMENTS \
 --rollup-config-out=$MONOREPO_BASE/rollup.json
```

**Comment by sbvegan** [comment]
Hey @boohyung, that data can be found here: https://github.com/ethereum-optimism/optimism-legacy/tree/develop/packages/migration-data/data

**Reply by theo-learner** [reply]
Hi. @sbvegan. Thanks for the reply. but I already got this so I want to know 'how to make a `ovm-allowance.json`. the allowance of OVM_ETH is disabled in https://github.com/ethereum-optimism/optimism/pull/1450. Why is it needed in Bedrock migration?

**Reply by sbvegan** [reply]
Ah I misread the question, sorry about that. I'll ask around and see if I can find out how to actually make the file. My assumption is that its necessary to track ERC20 versions of ETH from pre-bedrock to handle the update to native ETH. I'll get clarification and relay it over to you.

**Comment by smartcontracts** [comment, marked as answer]
`ovm-allowances.json` was generated by using a legacy (OVM 1.0) node to query for allowance events. The reason this file is needed is so that we could guarantee that *all* ETH balances were correctly migrated as part of the Bedrock migration. Guaranteeing this required that we confirmed that every storage slot within the `OVM_ETH` smart contract had been accounted for. Since this included allowances from the legacy OVM system, this meant that we also needed to generate a list of all allowances. The Bedrock migration script uses the combination of the `ovm-addresses`, `evm-addresses`, and `ovm-allowances` files to verify that all storage slots within the `OVM_ETH` contract are accounted for.

Generating this file again is difficult, so the Bedrock migration was designed in a way where the file is untrusted -- the Bedrock migration script verifies that the inputs are correct such that you can use a file given to you by some other party and check that it is correct. If the file missed any allowances or included any extra allowances, the Bedrock migration script would fail.
